mojo/stdlib/stdlib/gpu/compute/mma.mojo

```diff
@@ -48,18 +50,40 @@ fn get_amd_fp8_dtype() -> DType:
     """Gets the appropriate FP8 dtype for the current AMD GPU architecture.
 
     Returns:
-        `DType.float8_e4m3fn` for CDNA4+ GPUs, `DType.float8_e4m3fnuz` for older AMD GPUs.
+        - `DType.float8_e4m3fn` for CDNA4+ and RDNA4+ GPUs
+        - `DType.float8_e4m3fnuz` for CDNA1-3 GPUs
+        - `DType.invalid` for RDNA3 (no native FP8 support).
     """
-    return DType.float8_e4m3fn if _cdna_4_or_newer() else DType.float8_e4m3fnuz
+
+    @parameter
+    if _is_amd_rdna3():
+        return DType.invalid
+    elif _is_amd_rdna4():
+        return DType.float8_e4m3fn
+    else:
+        return (
+            DType.float8_e4m3fn if _cdna_4_or_newer() else DType.float8_e4m3fnuz
```

Comment:
A similar pattern to the one I point out in the matching function below could be applied here to simplify this conditional check.